theano/tensor/tests/test_nlinalg.py

```diff
@@ -429,6 +422,9 @@ def test_trace():
         trace(xx)
     except TypeError:
         ok = True
+    except ValueError:
```

Comment:
same here.

Reply:
Yes, because the `basic.py` implementation throws a `ValueError` instead of a type error. 
The fix would either be to:
- Add the exception for `ValueError`
- Change the exception to `TypeError`

Whichever you guys think is more appropriate.

Reply:
I think the `ValueError` is correct. The old code where the `TypeError` was raised is going to be removed so I think the `TypeError` should be removed. Is it raised any other place than line 225 of the old code?